upf/flowtable_init.c

```diff
@@ -80,6 +83,14 @@ flowtable_init_cpu (flowtable_main_t * fm, flowtable_main_per_cpu_t * fmt)
   pool_validate_index (fmt->timers, TIMER_MAX_LIFETIME - 1);
   upf_debug ("POOL SIZE %u", pool_elts (fmt->timers));
 
+  /* Init flows counter per cpu */
+  upf_counter_lock (gtm);
+  vlib_validate_simple_counter (&gtm->upf_simple_counters[UPF_FLOW_COUNTER],
+				fmt - fm->per_cpu);
+  vlib_zero_simple_counter (&gtm->upf_simple_counters[UPF_FLOW_COUNTER],
+			    fmt - fm->per_cpu);
+  upf_counter_unlock (gtm);
+
```

Comment:
putting the counters in flowtable_main_t might make more sense. simple_counters are per CPU thread, repeating that by using the index and CPU id does not make sense.

Or is there a specific reason you want to read the per CPU counters?

Reply:
I've just following your suggestion to make it per-cpu, but I suppose it was only suitable for gauges.
Keeping in mind that counters are per-cpu vectored more logical is to have in in flowtable_main for sure.
Will be changed.